packages/salesforcedx-utils-vscode/test/jest/telemetry/telemetry.test.ts

```diff
@@ -91,4 +91,68 @@ describe('Telemetry', () => {
       expect(result).toBe(telemetryService.extensionName);
     });
   });
+
+  describe('getUserId', () => {
+    let telemetryService: TelemetryService;
+    let executeCliTelemetrySpy: jest.SpyInstance;
+    let getRandomUserIdSpy: jest.SpyInstance;
+    let fakeGet: jest.SpyInstance;
+    let fakeUpdate: jest.SpyInstance;
+
+    const fakeCliTelemetryData = {
+      result: {
+        enabled: true,
+        cliId: 'rdccvghuhbnjiokmnmklop'
+      }
+    };
+    const fakeCliIdUndefined = {
+      result: {
+        enabled: true
+      }
+    };
+    const randomId = 'sedtrfyv234dd';
+    const globalTelemetryUserId = 'e45rdcfy7ygvhu8uhbnjiiugdrgb';
+
+    beforeEach(() => {
+      telemetryService = new TelemetryService();
+      fakeGet = jest.fn();
+      fakeUpdate = jest.fn();
+      (telemetryService as any).extensionContext = { globalState: { get: fakeGet, update: fakeUpdate } };
+      executeCliTelemetrySpy = jest.spyOn((telemetryService as any), 'executeCliTelemetry');
+      getRandomUserIdSpy = jest.spyOn((telemetryService as any), 'getRandomUserId');
+    });
+
+    it('should return cliId when telemetryUserId is undefined in global state', async () => {
+      fakeGet.mockReturnValueOnce(undefined);
+      fakeUpdate.mockResolvedValueOnce(undefined);
+
+      executeCliTelemetrySpy.mockResolvedValueOnce(JSON.stringify(fakeCliTelemetryData));
+
+      const uId = await telemetryService.getUserId();
+
+      expect(uId).toBe(fakeCliTelemetryData.result.cliId);
+      expect(executeCliTelemetrySpy).toHaveBeenCalled();
+    });
+
+    it('should generate random userId when telemetryUserId as well as cliId is undefined', async () => {
+      executeCliTelemetrySpy.mockResolvedValueOnce(JSON.stringify(fakeCliIdUndefined));
+      getRandomUserIdSpy.mockResolvedValueOnce(randomId);
+
+      const uId = await telemetryService.getUserId();
+
+      expect(getRandomUserIdSpy).toHaveBeenCalled();
+      expect(uId).toBe(randomId);
+    });
+
```

Comment:
The following test will cover the catch condition in the promise chain. 

```suggestion
   it('should generate random userId when cli command throws', async () => {
      executeCliTelemetrySpy.mockRejectedValueOnce(new Error('no cli for you'));
      getRandomUserIdSpy.mockResolvedValueOnce(randomId);

      const uId = await telemetryService.getUserId();

      expect(getRandomUserIdSpy).toHaveBeenCalled();
      expect(uId).toBe(randomId);
    });
```